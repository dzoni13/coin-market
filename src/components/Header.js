import React, { Component }            from 'react';
import { Navbar, Nav, NavItem, Badge } from 'react-bootstrap';

class Header extends Component {
  render() {
    return(
      <Navbar>
        <Navbar.Header>
          <Navbar.Brand>
            <Navbar.Link href="/"> Cryptocurrencies <Badge>100</Badge> </Navbar.Link>
          </Navbar.Brand>
        </Navbar.Header>
        <Nav>
          <NavItem href="/settings"> Settings </NavItem>
        </Nav>
      </Navbar>
    );
  }
}

export default Header;
