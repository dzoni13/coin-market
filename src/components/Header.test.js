import React        from 'react';
import { shallow }  from 'enzyme';
import Header       from './Header';

const header = shallow(<Header />);

describe('Header', () => {
  it('renders correctly', () => {
    expect(header).toMatchSnapshot();
  });
});
