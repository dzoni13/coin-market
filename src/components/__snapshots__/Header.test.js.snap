// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsStyle": "default",
      "children": Array [
        <NavbarHeader>
          <NavbarBrand>
                    <NavbarLink
                              componentClass="a"
                              href="/"
                              pullLeft={false}
                              pullRight={false}
                    >
                               Cryptocurrencies 
                              <Badge
                                        bsClass="badge"
                                        pullRight={false}
                              >
                                        100
                              </Badge>
                               
                    </NavbarLink>
          </NavbarBrand>
</NavbarHeader>,
        <Nav
          bsClass="nav"
          justified={false}
          pullLeft={false}
          pullRight={false}
          stacked={false}
>
          <NavItem
                    active={false}
                    disabled={false}
                    href="/settings"
          >
                     Settings 
          </NavItem>
</Nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <NavbarBrand>
            <NavbarLink
                        componentClass="a"
                        href="/"
                        pullLeft={false}
                        pullRight={false}
            >
                         Cryptocurrencies 
                        <Badge
                                    bsClass="badge"
                                    pullRight={false}
                        >
                                    100
                        </Badge>
                         
            </NavbarLink>
</NavbarBrand>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <NavbarLink
              componentClass="a"
              href="/"
              pullLeft={false}
              pullRight={false}
>
               Cryptocurrencies 
              <Badge
                            bsClass="badge"
                            pullRight={false}
              >
                            100
              </Badge>
               
</NavbarLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                " Cryptocurrencies ",
                <Badge
                  bsClass="badge"
                  pullRight={false}
>
                  100
</Badge>,
                " ",
              ],
              "componentClass": "a",
              "href": "/",
              "pullLeft": false,
              "pullRight": false,
            },
            "ref": null,
            "rendered": Array [
              " Cryptocurrencies ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "badge",
                  "children": "100",
                  "pullRight": false,
                },
                "ref": null,
                "rendered": "100",
                "type": [Function],
              },
              " ",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "nav",
          "children": <NavItem
            active={false}
            disabled={false}
            href="/settings"
>
             Settings 
</NavItem>,
          "justified": false,
          "pullLeft": false,
          "pullRight": false,
          "stacked": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": " Settings ",
            "disabled": false,
            "href": "/settings",
          },
          "ref": null,
          "rendered": " Settings ",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "default",
        "children": Array [
          <NavbarHeader>
            <NavbarBrand>
                        <NavbarLink
                                    componentClass="a"
                                    href="/"
                                    pullLeft={false}
                                    pullRight={false}
                        >
                                     Cryptocurrencies 
                                    <Badge
                                                bsClass="badge"
                                                pullRight={false}
                                    >
                                                100
                                    </Badge>
                                     
                        </NavbarLink>
            </NavbarBrand>
</NavbarHeader>,
          <Nav
            bsClass="nav"
            justified={false}
            pullLeft={false}
            pullRight={false}
            stacked={false}
>
            <NavItem
                        active={false}
                        disabled={false}
                        href="/settings"
            >
                         Settings 
            </NavItem>
</Nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <NavbarBrand>
              <NavbarLink
                            componentClass="a"
                            href="/"
                            pullLeft={false}
                            pullRight={false}
              >
                             Cryptocurrencies 
                            <Badge
                                          bsClass="badge"
                                          pullRight={false}
                            >
                                          100
                            </Badge>
                             
              </NavbarLink>
</NavbarBrand>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <NavbarLink
                componentClass="a"
                href="/"
                pullLeft={false}
                pullRight={false}
>
                 Cryptocurrencies 
                <Badge
                                bsClass="badge"
                                pullRight={false}
                >
                                100
                </Badge>
                 
</NavbarLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " Cryptocurrencies ",
                  <Badge
                    bsClass="badge"
                    pullRight={false}
>
                    100
</Badge>,
                  " ",
                ],
                "componentClass": "a",
                "href": "/",
                "pullLeft": false,
                "pullRight": false,
              },
              "ref": null,
              "rendered": Array [
                " Cryptocurrencies ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "badge",
                    "children": "100",
                    "pullRight": false,
                  },
                  "ref": null,
                  "rendered": "100",
                  "type": [Function],
                },
                " ",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "nav",
            "children": <NavItem
              active={false}
              disabled={false}
              href="/settings"
>
               Settings 
</NavItem>,
            "justified": false,
            "pullLeft": false,
            "pullRight": false,
            "stacked": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "children": " Settings ",
              "disabled": false,
              "href": "/settings",
            },
            "ref": null,
            "rendered": " Settings ",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
